src/components/HistoryPage.vue

```diff
@@ -20,20 +22,77 @@
     </div>
 
     <br />
+    <ModalComponent v-if="showModal" @close="showModal = !showModal">
```

Comment:
I feel like that modal should be placed in a different file to keep it easier to overview, let's see what K thinks about it.

Reply:
I agree, we should make a new component for it for future readability reasons.